mods/tnt/init.lua

```diff
@@ -384,7 +349,73 @@ minetest.register_craft({
 	}
 })
 
+function tnt.register_tnt(def)
+	local name = ""
+	if not def.name:find(':') then
+		name = "tnt:"..def.name
+	else
+		name = def.name
+		def.name:sub(def.name:find(':')+1)
```

Comment:
Still an ugly way for trimming a substring. Use `def.name:match(":([%w_]+)")` instead.

Better readable for any competent Lua programmer.